wsd/FileServer.cpp

```diff
@@ -458,15 +458,27 @@ bool FileServerRequestHandler::isAdminLoggedIn(const HTTPRequest& request,
             }
 
             std::streamsize size = request.getContentLength();
-            char buffer[size];
-            message.read(buffer, size);
-            LocalFileInfo::fileInfoVec[i].fileLastModifiedTime = std::chrono::system_clock::now();
-
             std::ofstream outfile;
             outfile.open(LocalFileInfo::fileInfoVec[i].localPath, std::ofstream::binary);
-            outfile.write(buffer, size);
+            constexpr std::streamsize bufferSize = 1024 * 1024;
+            std::unique_ptr<char[]> buffer(new char[bufferSize]);
+            if (size > bufferSize)
+            {
+                int numberOfChunks = (size / bufferSize) + 1;
+                for (int j=0; j<numberOfChunks; j++) {
+                    message.read(buffer.get(), bufferSize);
+                    outfile.write(buffer.get(), size);
```

Comment:
The outfile.write(buffer.get(), size); is strange - didn't you mean to use bufferSize there?  But with that, I fear this will enlarge the file - it won't stop on the exact amount, but a multiply of bufferSize? 

You should either use tellg() or gcount() somewhere - to also handle the case when read() reads less (not sure if that can happen - but normally when implementing a buffered read, that is something that has to be considered).

Fixing the above will also help you to get rid of the if (size > bufferSize) ... else ... I think, to make the code more compact...